src/weights.jl

```diff
@@ -1,9 +1,9 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+immutable WeightVec{T, V<:RealVector, S} <: RealVector{T}
```

Comment:
Is it really useful to allow `typeof(sum)` and `eltype(values)` to be different? Having three parameters for such a simple type seems overly complicated.